packages/kernel/src/shared-worker.ts

```diff
@@ -0,0 +1,52 @@
+import { startWorkerEnv } from "./worker-runtime";
+import type { InMessage, OutMessage, ReplyMessage } from "./types";
+
+declare class SharedWorkerGlobalScope {
+  onconnect:
+    | ((
+        this: SharedWorkerGlobalScope,
+        ev: MessageEvent<InMessage | OutMessage | ReplyMessage>,
+      ) => void)
+    | null;
+}
+
+const ports = new Set<MessagePort>();
+
+(self as unknown as SharedWorkerGlobalScope).onconnect = (event) => {
+  const port = event.ports[0];
+  ports.add(port);
+
+  console.debug("SharedWorker: Setting up message handler for new port");
+  port.onmessage = startWorkerEnv(
+    "/node_modules/pyodide/pyodide.mjs",
+    (message, targetPort) => {
```

Comment:
Does `startWorkerEnv` acutally deals with the `targetPort` somehow? Looks like it's never used.